src/main/java/org/aksw/iguana/cc/query/handler/QueryHandler.java

```diff
@@ -238,7 +243,7 @@ private QuerySource createQuerySource(Path path) throws IOException {
 
     public QuerySelector getQuerySelectorInstance() {
         switch (config.order()) {
-            case LINEAR -> { return new LinearQuerySelector(queryList.size()); }
+            case LINEAR -> { return new LinearQuerySelector(queryList.size(), totalWorkerCount != 0 ? (queryList.size() * workerCount++) / totalWorkerCount : 0); }
```

Comment:
When would `totalWorkerCount != 0` be true? 
Should this not be `totalWorkerCount != 1`?

Reply:
`totalWorkerCount` is zero during testing and you would get a lot of division by zero errors, so I added that condition to the method. It would also make that method more robust, so I just added that condition instead of changing every failing test.